plugins/modules/identity/ipa/ipa_dnsrecord.py

```diff
@@ -166,29 +184,31 @@ def dnsrecord_find(self, zone_name, record_name):
 
     def dnsrecord_add(self, zone_name=None, record_name=None, details=None):
         item = dict(idnsname=record_name)
-        if details['record_type'] == 'A':
-            item.update(a_part_ip_address=details['record_value'])
-        elif details['record_type'] == 'AAAA':
-            item.update(aaaa_part_ip_address=details['record_value'])
-        elif details['record_type'] == 'A6':
-            item.update(a6_part_data=details['record_value'])
-        elif details['record_type'] == 'CNAME':
-            item.update(cname_part_hostname=details['record_value'])
-        elif details['record_type'] == 'DNAME':
-            item.update(dname_part_target=details['record_value'])
-        elif details['record_type'] == 'PTR':
-            item.update(ptr_part_hostname=details['record_value'])
-        elif details['record_type'] == 'TXT':
-            item.update(txtrecord=details['record_value'])
-        elif details['record_type'] == 'SRV':
-            item.update(srvrecord=details['record_value'])
-        elif details['record_type'] == 'MX':
-            item.update(mxrecord=details['record_value'])
 
         if details.get('record_ttl'):
             item.update(dnsttl=details['record_ttl'])
 
-        return self._post_json(method='dnsrecord_add', name=zone_name, item=item)
+        for value in details['record_value']:
```

Comment:
In line 272 you store `record_values` in `details`, and not `record_value`.
```suggestion
        for value in details['record_values']:
```

Reply:
ups - good found